{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Great, now that we discussed a little let's continue"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Given that the current approach utilized by the authors lacks reproducibility, we will explore an alternative method by leveraging nf-core pipelines for data analysis.\n",
    "\n",
    "Please explain, how we will achieve reproducibility for the course  with this approach.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will use the fetchngs pipeline from nf-core. The pipeline has been set-up to automatically download and process the raw FastQ files from public repositories."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You have successfully downloaded 2 of the fastq files we will use in our study.\n",
    "\n",
    "What is the next step if we want to first have a count table and check the quality of our fastq files? What is the pipeline called to do so?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "nf-core/rnaseq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Analyze the 2 files using an nf-core pipeline.\n",
    "\n",
    "What does this pipeline do? <br>\n",
    "The pipeline does all preprocessing of RNA-seq data (QC → alignment → counting → QC summary).\n",
    "\n",
    "Which are the main tools that will be used in the pipeline? <br>\n",
    "The main tools are: FastQC, TrimGalore, STAR/HISAT2/Salmon, featureCounts, RSeQC, Samtools, Picard, and MultiQC.\n",
    "\n",
    "<br>\n",
    "compared with the aper the nf-core/rnaseq covers everything up to the count matrix (alignment, QC, read counting).\n",
    "The paper’s workflow continues into statistical modeling and biological interpretation (factorial DESeq2 design, heatmaps, pathway analysis), which has to be done separately after running nf-core/rnaseq."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As all other nf-core pipelines, the chosen pipeline takes in a samplesheet as input.\n",
    "\n",
    "Use Python and pandas to create the samplesheet for your 2 samples. Feel free to make use of the table you created earlier today.\n",
    "\n",
    "Choose your sample names wisely, they must be the connection of the results to the metadata. If you can't find the sample in the metadata later, the analysis was useless."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\u001b[1m\u001b[38;5;232m\u001b[48;5;43m N E X T F L O W \u001b[0;2m  ~  \u001b[mversion 25.04.7\u001b[m\n",
      "\u001b[K\n",
      "Launching\u001b[35m `https://github.com/nf-core/rnaseq` \u001b[0;2m[\u001b[0;1;36mloving_descartes\u001b[0;2m] DSL2 - \u001b[36mrevision: \u001b[0;36m9738a2df42 [master]\u001b[m\n",
      "\u001b[K\n",
      "\n",
      "------------------------------------------------------\n",
      "                                        ,--./,-.\n",
      "        ___     __   __   __   ___     /,-._.--~'\n",
      "  |\\ | |__  __ /  ` /  \\ |__) |__         }  {\n",
      "  | \\| |       \\__, \\__/ |  \\ |___     \\`-._,-`-,\n",
      "                                        `._,._,'\n",
      "  nf-core/rnaseq 3.21.0\n",
      "------------------------------------------------------\n",
      "\u001b[1mInput/output options\u001b[0m\n",
      "  \u001b[0;34minput              : \u001b[0;32mSRFetch_results/samplesheet/samplesheet_fixed.csv\u001b[0m\n",
      "  \u001b[0;34moutdir             : \u001b[0;32mrnaSeq\u001b[0m\n",
      "\n",
      "\u001b[1mReference genome options\u001b[0m\n",
      "  \u001b[0;34mgenome             : \u001b[0;32mGRCm38\u001b[0m\n",
      "  \u001b[0;34mfasta              : \u001b[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Sequence/WholeGenomeFasta/genome.fa\u001b[0m\n",
      "  \u001b[0;34mgtf                : \u001b[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.gtf\u001b[0m\n",
      "  \u001b[0;34mgene_bed           : \u001b[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.bed\u001b[0m\n",
      "  \u001b[0;34mstar_index         : \u001b[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Sequence/STARIndex/\u001b[0m\n",
      "\n",
      "\u001b[1mUMI options\u001b[0m\n",
      "  \u001b[0;34mumi_discard_read   : \u001b[0;32m0\u001b[0m\n",
      "\n",
      "\u001b[1mAlignment options\u001b[0m\n",
      "  \u001b[0;34maligner            : \u001b[0;32mhisat2\u001b[0m\n",
      "\n",
      "\u001b[1mGeneric options\u001b[0m\n",
      "  \u001b[0;34mtrace_report_suffix: \u001b[0;32m2025-09-30_14-39-01\u001b[0m\n",
      "\n",
      "\u001b[1mCore Nextflow options\u001b[0m\n",
      "  \u001b[0;34mrevision           : \u001b[0;32mmaster\u001b[0m\n",
      "  \u001b[0;34mrunName            : \u001b[0;32mloving_descartes\u001b[0m\n",
      "  \u001b[0;34mcontainerEngine    : \u001b[0;32mdocker\u001b[0m\n",
      "  \u001b[0;34mlaunchDir          : \u001b[0;32m/Users/leoniewehnert/Desktop/computational-workflows-2025/notebooks/day_02\u001b[0m\n",
      "  \u001b[0;34mworkDir            : \u001b[0;32m/Users/leoniewehnert/Desktop/computational-workflows-2025/notebooks/day_02/work\u001b[0m\n",
      "  \u001b[0;34mprojectDir         : \u001b[0;32m/Users/leoniewehnert/.nextflow/assets/nf-core/rnaseq\u001b[0m\n",
      "  \u001b[0;34muserName           : \u001b[0;32mleoniewehnert\u001b[0m\n",
      "  \u001b[0;34mprofile            : \u001b[0;32mdocker\u001b[0m\n",
      "  \u001b[0;34mconfigFiles        : \u001b[0;32m/Users/leoniewehnert/.nextflow/assets/nf-core/rnaseq/nextflow.config\u001b[0m\n",
      "\n",
      "!! Only displaying parameters that differ from the pipeline defaults !!\n",
      "-\u001b[2m----------------------------------------------------\u001b[0m-\n",
      "* The pipeline\n",
      "    https://doi.org/10.5281/zenodo.1400710\n",
      "\n",
      "* The nf-core framework\n",
      "    https://doi.org/10.1038/s41587-020-0439-x\n",
      "\n",
      "* Software dependencies\n",
      "    https://github.com/nf-core/rnaseq/blob/master/CITATIONS.md\n",
      "\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OME:\u001b[mMAKE_TRANSCRIPTS_FASTA -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ENOME:\u001b[mCUSTOM_GETCHROMSIZES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…REPARE_GENOME:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ASEQ:RNASEQ:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001b[39m\u001b[K\n",
      "\u001b[33mWARN: Found the following unidentified headers in /Users/leoniewehnert/Desktop/computational-workflows-2025/notebooks/day_02/SRFetch_results/samplesheet/samplesheet_fixed.csv:\u001b[K\n",
      "\t- run_accession\u001b[K\n",
      "\t- experiment_accession\u001b[K\n",
      "\t- sample_accession\u001b[K\n",
      "\t- secondary_sample_accession\u001b[K\n",
      "\t- study_accession\u001b[K\n",
      "\t- secondary_study_accession\u001b[K\n",
      "\t- submission_accession\u001b[K\n",
      "\t- run_alias\u001b[K\n",
      "\t- experiment_alias\u001b[K\n",
      "\t- sample_alias\u001b[K\n",
      "\t- study_alias\u001b[K\n",
      "\t- library_layout\u001b[K\n",
      "\t- library_selection\u001b[K\n",
      "\t- library_source\u001b[K\n",
      "\t- library_strategy\u001b[K\n",
      "\t- library_name\u001b[K\n",
      "\t- instrument_model\u001b[K\n",
      "\t- instrument_platform\u001b[K\n",
      "\t- base_count\u001b[K\n",
      "\t- read_count\u001b[K\n",
      "\t- tax_id\u001b[K\n",
      "\t- scientific_name\u001b[K\n",
      "\t- sample_title\u001b[K\n",
      "\t- experiment_title\u001b[K\n",
      "\t- study_title\u001b[K\n",
      "\t- sample_description\u001b[K\n",
      "\t- fastq_md5\u001b[K\n",
      "\t- fastq_bytes\u001b[K\n",
      "\t- fastq_ftp\u001b[K\n",
      "\t- fastq_galaxy\u001b[K\n",
      "\t- fastq_aspera\u001b[K\n",
      "\u001b[39m\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OME:\u001b[mMAKE_TRANSCRIPTS_FASTA -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ENOME:\u001b[mCUSTOM_GETCHROMSIZES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…REPARE_GENOME:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ASEQ:RNASEQ:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_SETSTRANDEDNESS:\u001b[mCAT_FASTQ -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ER_SETSTRANDEDNESS:\u001b[mFQ_LINT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…UMITOOLS_TRIMGALORE:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OOLS_TRIMGALORE:\u001b[mTRIMGALORE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ESS:\u001b[mFQ_LINT_AFTER_TRIMMING -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mFQ_SUBSAMPLE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_QUANT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_ALIGN_HISAT2:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ATS_SAMTOOLS:\u001b[mSAMTOOLS_SORT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_STATS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_FLAGSTAT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_IDXSTATS -\u001b[K\n",
      "\u001b[3;2mPlus \u001b[1m8\u001b[0;3;2m more processes waiting for tasks…\u001b[m\u001b[K\n",
      "\u001b[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001b[39m\u001b[K\n",
      "\u001b[33mWARN: Found the following unidentified headers in /Users/leoniewehnert/Desktop/computational-workflows-2025/notebooks/day_02/SRFetch_results/samplesheet/samplesheet_fixed.csv:\u001b[K\n",
      "\t- run_accession\u001b[K\n",
      "\t- experiment_accession\u001b[K\n",
      "\t- sample_accession\u001b[K\n",
      "\t- secondary_sample_accession\u001b[K\n",
      "\t- study_accession\u001b[K\n",
      "\t- secondary_study_accession\u001b[K\n",
      "\t- submission_accession\u001b[K\n",
      "\t- run_alias\u001b[K\n",
      "\t- experiment_alias\u001b[K\n",
      "\t- sample_alias\u001b[K\n",
      "\t- study_alias\u001b[K\n",
      "\t- library_layout\u001b[K\n",
      "\t- library_selection\u001b[K\n",
      "\t- library_source\u001b[K\n",
      "\t- library_strategy\u001b[K\n",
      "\t- library_name\u001b[K\n",
      "\t- instrument_model\u001b[K\n",
      "\t- instrument_platform\u001b[K\n",
      "\t- base_count\u001b[K\n",
      "\t- read_count\u001b[K\n",
      "\t- tax_id\u001b[K\n",
      "\t- scientific_name\u001b[K\n",
      "\t- sample_title\u001b[K\n",
      "\t- experiment_title\u001b[K\n",
      "\t- study_title\u001b[K\n",
      "\t- sample_description\u001b[K\n",
      "\t- fastq_md5\u001b[K\n",
      "\t- fastq_bytes\u001b[K\n",
      "\t- fastq_ftp\u001b[K\n",
      "\t- fastq_galaxy\u001b[K\n",
      "\t- fastq_aspera\u001b[K\n",
      "\u001b[39m\u001b[K\n",
      "\u001b[56A\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OME:\u001b[mMAKE_TRANSCRIPTS_FASTA -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ENOME:\u001b[mCUSTOM_GETCHROMSIZES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…REPARE_GENOME:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ASEQ:RNASEQ:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_SETSTRANDEDNESS:\u001b[mCAT_FASTQ -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ER_SETSTRANDEDNESS:\u001b[mFQ_LINT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…UMITOOLS_TRIMGALORE:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OOLS_TRIMGALORE:\u001b[mTRIMGALORE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ESS:\u001b[mFQ_LINT_AFTER_TRIMMING -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mFQ_SUBSAMPLE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_QUANT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_ALIGN_HISAT2:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ATS_SAMTOOLS:\u001b[mSAMTOOLS_SORT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_STATS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_FLAGSTAT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_IDXSTATS -\u001b[K\n",
      "\u001b[3;2mPlus \u001b[1m23\u001b[0;3;2m more processes waiting for tasks…\u001b[m\u001b[K\n",
      "\u001b[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001b[39m\u001b[K\n",
      "\u001b[33mWARN: Found the following unidentified headers in /Users/leoniewehnert/Desktop/computational-workflows-2025/notebooks/day_02/SRFetch_results/samplesheet/samplesheet_fixed.csv:\u001b[K\n",
      "\t- run_accession\u001b[K\n",
      "\t- experiment_accession\u001b[K\n",
      "\t- sample_accession\u001b[K\n",
      "\t- secondary_sample_accession\u001b[K\n",
      "\t- study_accession\u001b[K\n",
      "\t- secondary_study_accession\u001b[K\n",
      "\t- submission_accession\u001b[K\n",
      "\t- run_alias\u001b[K\n",
      "\t- experiment_alias\u001b[K\n",
      "\t- sample_alias\u001b[K\n",
      "\t- study_alias\u001b[K\n",
      "\t- library_layout\u001b[K\n",
      "\t- library_selection\u001b[K\n",
      "\t- library_source\u001b[K\n",
      "\t- library_strategy\u001b[K\n",
      "\t- library_name\u001b[K\n",
      "\t- instrument_model\u001b[K\n",
      "\t- instrument_platform\u001b[K\n",
      "\t- base_count\u001b[K\n",
      "\t- read_count\u001b[K\n",
      "\t- tax_id\u001b[K\n",
      "\t- scientific_name\u001b[K\n",
      "\t- sample_title\u001b[K\n",
      "\t- experiment_title\u001b[K\n",
      "\t- study_title\u001b[K\n",
      "\t- sample_description\u001b[K\n",
      "\t- fastq_md5\u001b[K\n",
      "\t- fastq_bytes\u001b[K\n",
      "\t- fastq_ftp\u001b[K\n",
      "\t- fastq_galaxy\u001b[K\n",
      "\t- fastq_aspera\u001b[K\n",
      "\u001b[39m\u001b[K\n",
      "\u001b[56A\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OME:\u001b[mMAKE_TRANSCRIPTS_FASTA -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ENOME:\u001b[mCUSTOM_GETCHROMSIZES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…REPARE_GENOME:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ASEQ:RNASEQ:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_SETSTRANDEDNESS:\u001b[mCAT_FASTQ -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ER_SETSTRANDEDNESS:\u001b[mFQ_LINT\u001b[2m |\u001b[m 0 of 2\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…UMITOOLS_TRIMGALORE:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OOLS_TRIMGALORE:\u001b[mTRIMGALORE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ESS:\u001b[mFQ_LINT_AFTER_TRIMMING -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mFQ_SUBSAMPLE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_QUANT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_ALIGN_HISAT2:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ATS_SAMTOOLS:\u001b[mSAMTOOLS_SORT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_STATS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_FLAGSTAT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_IDXSTATS -\u001b[K\n",
      "\u001b[3;2mPlus \u001b[1m24\u001b[0;3;2m more processes waiting for tasks…\u001b[m\u001b[K\n",
      "\u001b[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001b[39m\u001b[K\n",
      "\u001b[33mWARN: Found the following unidentified headers in /Users/leoniewehnert/Desktop/computational-workflows-2025/notebooks/day_02/SRFetch_results/samplesheet/samplesheet_fixed.csv:\u001b[K\n",
      "\t- run_accession\u001b[K\n",
      "\t- experiment_accession\u001b[K\n",
      "\t- sample_accession\u001b[K\n",
      "\t- secondary_sample_accession\u001b[K\n",
      "\t- study_accession\u001b[K\n",
      "\t- secondary_study_accession\u001b[K\n",
      "\t- submission_accession\u001b[K\n",
      "\t- run_alias\u001b[K\n",
      "\t- experiment_alias\u001b[K\n",
      "\t- sample_alias\u001b[K\n",
      "\t- study_alias\u001b[K\n",
      "\t- library_layout\u001b[K\n",
      "\t- library_selection\u001b[K\n",
      "\t- library_source\u001b[K\n",
      "\t- library_strategy\u001b[K\n",
      "\t- library_name\u001b[K\n",
      "\t- instrument_model\u001b[K\n",
      "\t- instrument_platform\u001b[K\n",
      "\t- base_count\u001b[K\n",
      "\t- read_count\u001b[K\n",
      "\t- tax_id\u001b[K\n",
      "\t- scientific_name\u001b[K\n",
      "\t- sample_title\u001b[K\n",
      "\t- experiment_title\u001b[K\n",
      "\t- study_title\u001b[K\n",
      "\t- sample_description\u001b[K\n",
      "\t- fastq_md5\u001b[K\n",
      "\t- fastq_bytes\u001b[K\n",
      "\t- fastq_ftp\u001b[K\n",
      "\t- fastq_galaxy\u001b[K\n",
      "\t- fastq_aspera\u001b[K\n",
      "\u001b[39m\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FQ_LINT (SRX19144486)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process requirement exceeds available memory -- req: 12 GB; avail: 8 GB\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  fq lint \\\u001b[K\n",
      "      --disable-validator P001 \\\u001b[K\n",
      "      SRX19144486_SRR23195516_1.fastq.gz SRX19144486_SRR23195516_2.fastq.gz > SRX19144486.fq_lint.txt\u001b[K\n",
      "  \u001b[K\n",
      "  cat <<-END_VERSIONS > versions.yml\u001b[K\n",
      "  \"NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FQ_LINT\":\u001b[K\n",
      "      fq: $(echo $(fq lint --version | sed 's/fq-lint //g'))\u001b[K\n",
      "  END_VERSIONS\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  -\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  (empty)\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  /Users/leoniewehnert/Desktop/computational-workflows-2025/notebooks/day_02/work/68/ead980e7f4f61634d5500d3d3ea79b\u001b[K\n",
      "\u001b[K\n",
      "Container:\u001b[K\n",
      "  quay.io/biocontainers/fq:0.12.0--h9ee0642_0\u001b[K\n",
      "\u001b[K\n",
      "Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[88A\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OME:\u001b[mMAKE_TRANSCRIPTS_FASTA -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ENOME:\u001b[mCUSTOM_GETCHROMSIZES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…REPARE_GENOME:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ASEQ:RNASEQ:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_SETSTRANDEDNESS:\u001b[mCAT_FASTQ -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ER_SETSTRANDEDNESS:\u001b[mFQ_LINT\u001b[2m |\u001b[m 0 of 2\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…UMITOOLS_TRIMGALORE:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OOLS_TRIMGALORE:\u001b[mTRIMGALORE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ESS:\u001b[mFQ_LINT_AFTER_TRIMMING -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mFQ_SUBSAMPLE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_QUANT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_ALIGN_HISAT2:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ATS_SAMTOOLS:\u001b[mSAMTOOLS_SORT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_STATS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_FLAGSTAT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_IDXSTATS -\u001b[K\n",
      "\u001b[3;2mPlus \u001b[1m24\u001b[0;3;2m more processes waiting for tasks…\u001b[m\u001b[K\n",
      "Execution cancelled -- Finishing pending tasks before exit\u001b[K\n",
      "\u001b[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001b[39m\u001b[K\n",
      "\u001b[33mWARN: Found the following unidentified headers in /Users/leoniewehnert/Desktop/computational-workflows-2025/notebooks/day_02/SRFetch_results/samplesheet/samplesheet_fixed.csv:\u001b[K\n",
      "\t- run_accession\u001b[K\n",
      "\t- experiment_accession\u001b[K\n",
      "\t- sample_accession\u001b[K\n",
      "\t- secondary_sample_accession\u001b[K\n",
      "\t- study_accession\u001b[K\n",
      "\t- secondary_study_accession\u001b[K\n",
      "\t- submission_accession\u001b[K\n",
      "\t- run_alias\u001b[K\n",
      "\t- experiment_alias\u001b[K\n",
      "\t- sample_alias\u001b[K\n",
      "\t- study_alias\u001b[K\n",
      "\t- library_layout\u001b[K\n",
      "\t- library_selection\u001b[K\n",
      "\t- library_source\u001b[K\n",
      "\t- library_strategy\u001b[K\n",
      "\t- library_name\u001b[K\n",
      "\t- instrument_model\u001b[K\n",
      "\t- instrument_platform\u001b[K\n",
      "\t- base_count\u001b[K\n",
      "\t- read_count\u001b[K\n",
      "\t- tax_id\u001b[K\n",
      "\t- scientific_name\u001b[K\n",
      "\t- sample_title\u001b[K\n",
      "\t- experiment_title\u001b[K\n",
      "\t- study_title\u001b[K\n",
      "\t- sample_description\u001b[K\n",
      "\t- fastq_md5\u001b[K\n",
      "\t- fastq_bytes\u001b[K\n",
      "\t- fastq_ftp\u001b[K\n",
      "\t- fastq_galaxy\u001b[K\n",
      "\t- fastq_aspera\u001b[K\n",
      "\u001b[39m\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FQ_LINT (SRX19144486)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process requirement exceeds available memory -- req: 12 GB; avail: 8 GB\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  fq lint \\\u001b[K\n",
      "      --disable-validator P001 \\\u001b[K\n",
      "      SRX19144486_SRR23195516_1.fastq.gz SRX19144486_SRR23195516_2.fastq.gz > SRX19144486.fq_lint.txt\u001b[K\n",
      "  \u001b[K\n",
      "  cat <<-END_VERSIONS > versions.yml\u001b[K\n",
      "  \"NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FQ_LINT\":\u001b[K\n",
      "      fq: $(echo $(fq lint --version | sed 's/fq-lint //g'))\u001b[K\n",
      "  END_VERSIONS\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  -\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  (empty)\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  /Users/leoniewehnert/Desktop/computational-workflows-2025/notebooks/day_02/work/68/ead980e7f4f61634d5500d3d3ea79b\u001b[K\n",
      "\u001b[K\n",
      "Container:\u001b[K\n",
      "  quay.io/biocontainers/fq:0.12.0--h9ee0642_0\u001b[K\n",
      "\u001b[K\n",
      "Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[31mERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[92A\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OME:\u001b[mMAKE_TRANSCRIPTS_FASTA -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ENOME:\u001b[mCUSTOM_GETCHROMSIZES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…REPARE_GENOME:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ASEQ:RNASEQ:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_SETSTRANDEDNESS:\u001b[mCAT_FASTQ -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m68/ead980\u001b[0;2m] \u001b[0;2mNFC…NESS:\u001b[mFQ_LINT\u001b[33;2m (\u001b[0;33mSRX19144486\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 2\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…UMITOOLS_TRIMGALORE:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OOLS_TRIMGALORE:\u001b[mTRIMGALORE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ESS:\u001b[mFQ_LINT_AFTER_TRIMMING -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mFQ_SUBSAMPLE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_QUANT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_ALIGN_HISAT2:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ATS_SAMTOOLS:\u001b[mSAMTOOLS_SORT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_STATS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_FLAGSTAT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_IDXSTATS -\u001b[K\n",
      "\u001b[3;2mPlus \u001b[1m24\u001b[0;3;2m more processes waiting for tasks…\u001b[m\u001b[K\n",
      "Execution cancelled -- Finishing pending tasks before exit\u001b[K\n",
      "-\u001b[0;35m[nf-core/rnaseq]\u001b[0;31m Pipeline completed with errors\u001b[0m-\u001b[K\n",
      "\u001b[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001b[39m\u001b[K\n",
      "\u001b[33mWARN: Found the following unidentified headers in /Users/leoniewehnert/Desktop/computational-workflows-2025/notebooks/day_02/SRFetch_results/samplesheet/samplesheet_fixed.csv:\u001b[K\n",
      "\t- run_accession\u001b[K\n",
      "\t- experiment_accession\u001b[K\n",
      "\t- sample_accession\u001b[K\n",
      "\t- secondary_sample_accession\u001b[K\n",
      "\t- study_accession\u001b[K\n",
      "\t- secondary_study_accession\u001b[K\n",
      "\t- submission_accession\u001b[K\n",
      "\t- run_alias\u001b[K\n",
      "\t- experiment_alias\u001b[K\n",
      "\t- sample_alias\u001b[K\n",
      "\t- study_alias\u001b[K\n",
      "\t- library_layout\u001b[K\n",
      "\t- library_selection\u001b[K\n",
      "\t- library_source\u001b[K\n",
      "\t- library_strategy\u001b[K\n",
      "\t- library_name\u001b[K\n",
      "\t- instrument_model\u001b[K\n",
      "\t- instrument_platform\u001b[K\n",
      "\t- base_count\u001b[K\n",
      "\t- read_count\u001b[K\n",
      "\t- tax_id\u001b[K\n",
      "\t- scientific_name\u001b[K\n",
      "\t- sample_title\u001b[K\n",
      "\t- experiment_title\u001b[K\n",
      "\t- study_title\u001b[K\n",
      "\t- sample_description\u001b[K\n",
      "\t- fastq_md5\u001b[K\n",
      "\t- fastq_bytes\u001b[K\n",
      "\t- fastq_ftp\u001b[K\n",
      "\t- fastq_galaxy\u001b[K\n",
      "\t- fastq_aspera\u001b[K\n",
      "\u001b[39m\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FQ_LINT (SRX19144486)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process requirement exceeds available memory -- req: 12 GB; avail: 8 GB\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  fq lint \\\u001b[K\n",
      "      --disable-validator P001 \\\u001b[K\n",
      "      SRX19144486_SRR23195516_1.fastq.gz SRX19144486_SRR23195516_2.fastq.gz > SRX19144486.fq_lint.txt\u001b[K\n",
      "  \u001b[K\n",
      "  cat <<-END_VERSIONS > versions.yml\u001b[K\n",
      "  \"NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FQ_LINT\":\u001b[K\n",
      "      fq: $(echo $(fq lint --version | sed 's/fq-lint //g'))\u001b[K\n",
      "  END_VERSIONS\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  -\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  (empty)\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  /Users/leoniewehnert/Desktop/computational-workflows-2025/notebooks/day_02/work/68/ead980e7f4f61634d5500d3d3ea79b\u001b[K\n",
      "\u001b[K\n",
      "Container:\u001b[K\n",
      "  quay.io/biocontainers/fq:0.12.0--h9ee0642_0\u001b[K\n",
      "\u001b[K\n",
      "Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[31mERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[93A\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OME:\u001b[mMAKE_TRANSCRIPTS_FASTA -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ENOME:\u001b[mCUSTOM_GETCHROMSIZES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…REPARE_GENOME:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ASEQ:RNASEQ:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_SETSTRANDEDNESS:\u001b[mCAT_FASTQ -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m68/ead980\u001b[0;2m] \u001b[0;2mNFC…NESS:\u001b[mFQ_LINT\u001b[33;2m (\u001b[0;33mSRX19144486\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 2\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…UMITOOLS_TRIMGALORE:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OOLS_TRIMGALORE:\u001b[mTRIMGALORE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ESS:\u001b[mFQ_LINT_AFTER_TRIMMING -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mFQ_SUBSAMPLE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_QUANT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_ALIGN_HISAT2:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ATS_SAMTOOLS:\u001b[mSAMTOOLS_SORT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_STATS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_FLAGSTAT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_IDXSTATS -\u001b[K\n",
      "\u001b[3;2mPlus \u001b[1m24\u001b[0;3;2m more processes waiting for tasks…\u001b[m\u001b[K\n",
      "Execution cancelled -- Finishing pending tasks before exit\u001b[K\n",
      "-\u001b[0;35m[nf-core/rnaseq]\u001b[0;31m Pipeline completed with errors\u001b[0m-\u001b[K\n",
      "\u001b[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001b[39m\u001b[K\n",
      "\u001b[33mWARN: Found the following unidentified headers in /Users/leoniewehnert/Desktop/computational-workflows-2025/notebooks/day_02/SRFetch_results/samplesheet/samplesheet_fixed.csv:\u001b[K\n",
      "\t- run_accession\u001b[K\n",
      "\t- experiment_accession\u001b[K\n",
      "\t- sample_accession\u001b[K\n",
      "\t- secondary_sample_accession\u001b[K\n",
      "\t- study_accession\u001b[K\n",
      "\t- secondary_study_accession\u001b[K\n",
      "\t- submission_accession\u001b[K\n",
      "\t- run_alias\u001b[K\n",
      "\t- experiment_alias\u001b[K\n",
      "\t- sample_alias\u001b[K\n",
      "\t- study_alias\u001b[K\n",
      "\t- library_layout\u001b[K\n",
      "\t- library_selection\u001b[K\n",
      "\t- library_source\u001b[K\n",
      "\t- library_strategy\u001b[K\n",
      "\t- library_name\u001b[K\n",
      "\t- instrument_model\u001b[K\n",
      "\t- instrument_platform\u001b[K\n",
      "\t- base_count\u001b[K\n",
      "\t- read_count\u001b[K\n",
      "\t- tax_id\u001b[K\n",
      "\t- scientific_name\u001b[K\n",
      "\t- sample_title\u001b[K\n",
      "\t- experiment_title\u001b[K\n",
      "\t- study_title\u001b[K\n",
      "\t- sample_description\u001b[K\n",
      "\t- fastq_md5\u001b[K\n",
      "\t- fastq_bytes\u001b[K\n",
      "\t- fastq_ftp\u001b[K\n",
      "\t- fastq_galaxy\u001b[K\n",
      "\t- fastq_aspera\u001b[K\n",
      "\u001b[39m\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FQ_LINT (SRX19144486)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process requirement exceeds available memory -- req: 12 GB; avail: 8 GB\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  fq lint \\\u001b[K\n",
      "      --disable-validator P001 \\\u001b[K\n",
      "      SRX19144486_SRR23195516_1.fastq.gz SRX19144486_SRR23195516_2.fastq.gz > SRX19144486.fq_lint.txt\u001b[K\n",
      "  \u001b[K\n",
      "  cat <<-END_VERSIONS > versions.yml\u001b[K\n",
      "  \"NFCORE_RNASEQ:RNASEQ:FASTQ_QC_TRIM_FILTER_SETSTRANDEDNESS:FQ_LINT\":\u001b[K\n",
      "      fq: $(echo $(fq lint --version | sed 's/fq-lint //g'))\u001b[K\n",
      "  END_VERSIONS\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  -\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  (empty)\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  /Users/leoniewehnert/Desktop/computational-workflows-2025/notebooks/day_02/work/68/ead980e7f4f61634d5500d3d3ea79b\u001b[K\n",
      "\u001b[K\n",
      "Container:\u001b[K\n",
      "  quay.io/biocontainers/fq:0.12.0--h9ee0642_0\u001b[K\n",
      "\u001b[K\n",
      "Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[31mERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Mus_musculus/Ensembl/GRCm38/Sequence/WholeGenomeFasta/genome.fa\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.gtf\n",
      "Waiting for file transfers to complete (2 files)\n",
      "One more CTRL+C to force exit\n"
     ]
    }
   ],
   "source": [
    "# post here the command you used to run nf-core/rnaseq\n",
    "!nextflow run \\\n",
    "    nf-core/rnaseq \\\n",
    "    --input \"SRFetch_results/samplesheet/samplesheet_fixed.csv\" \\\n",
    "    --aligner hisat2 \\\n",
    "    --outdir rnaSeq \\\n",
    "    --genome GRCm38 \\\n",
    "    -profile docker"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I could not run this command due to memory issues"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Explain all the parameters you set and why you set them in this way.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The input directory points to the samplesheet provided by the nf-core/fetchngs pipeline (fixed based on strandedness). The aligner was chosen based on the paper. The output directory was set to collect the output and the genome was set to the genome of Mus Musculus."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Browsing the results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How did the pipeline perform?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Explain the quality control steps. Are you happy with the quality and why. If not, why not.\n",
    "Please give additional information on : \n",
    "- ribosomal rRNA\n",
    "- Duplication\n",
    "- GC content\n",
    "\n",
    "What are the possible steps that could lead to poorer results?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Would you exclude any samples? If yes, which and why?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "What would you now do to continue the experiment? What are the scientists trying to figure out? Which packages on R or python would you use?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Nextflow_Prak",
   "language": "python",
   "name": "myenv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
